### ISSUE TITLE
|= does not work with utf-8

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2807]
Migrated from [rt.perl.org#4604](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=4604) (status was 'resolved')

Searchable as RT4604$

**Comment by p5pRT** [comment]
### From pcg@goof.com

#### Created by pcg@goof.com



This Program&#8203;:

&nbsp;   $a = "/tmp/papp\.\.\.";
&nbsp;   $b = '/tmp/papp\.top';

&nbsp;   use Encode;
&nbsp;   Encode&#8203;::on\_utf8\($a\);

&nbsp;   $a |= $b;

&nbsp;   print $a;

Prints

"/tmp/papp\.~op"

because the |= operator does strange things \(it seems that dots trigger
this behaviour ;\)\. Testes with the perl evrsion below and DEVEL7523\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.7.0:

Configured by root at Sun Oct 29 21:53:30 CET 2000.

Summary of my perl5 (revision 5.0 version 7 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.2, archname=i686-linux
    uname='linux cerebro 2.2.17 #1 smp mon oct 16 00:47:15 cest 2000 i686 unknown '
    config_args=''
    hint=previous, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=undef usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='gcc', ccflags ='-I/usr/local/include -I/usr/app/include -I/opt/include',
    optimize='-g -Os -fno-omit-frame-pointer -mpentiumpro',
    cppflags='-I/usr/local/include -I/usr/app/include -I/opt/include'
    ccversion='', gccversion='pgcc-2.95.2 19991024 (release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-L/usr/local/lib -L/opt/lib'
    libpth=/usr/local/lib /lib /usr/lib /opt/lib
    libs=-lc -ldl -lm -lcrypt
    perllibs=-lc -ldl -lm -lcrypt
    libc=/lib/libc-2.1.3.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib -L/opt/lib'

Locally applied patches:
    DEVEL7481


@INC for perl v5.7.0:
    /usr/app/lib/perl5
    /usr/app/lib/perl5
    /usr/app/lib/perl5
    /usr/app/lib/perl5
    .


Environment for perl v5.7.0:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=de_DE
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/root/s:/bin:/usr/bin:/usr/app/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/usr/app/bin:/usr/app/sbin:/usr/X11/bin:/opt/jdk118/bin:/opt/bin:/opt/sbin:.:/root/cc/dejagnu/bin
    PERLDB_OPTS=ornaments=0
    PERL_BADLANG (unset)
    SHELL=/bin/bash



```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Sun\, Nov 05\, 2000 at 11&#8203;:09&#8203;:19PM \-0000\, perlbugtron@&#8203;bugs\.perl\.org wrote&#8203;:
> A Bug ID \(20001105\.014\) has been assigned and is shown in the subject of this email\.  

Unexpectedly\, perl\.org passed my message thru when it shouldn't have\. Please
disregard/close/kill the previous bug report\, as it was obviously bogus\.

**Comment by p5pRT** [comment]
### From @tux

On Mon\, 6 Nov 2000 00&#8203;:11&#8203;:14 \+0100\, Marc Lehmann \<pcg@&#8203;goof\.com> wrote&#8203;:
> On Sun\, Nov 05\, 2000 at 11&#8203;:09&#8203;:19PM \-0000\, perlbugtron@&#8203;bugs\.perl\.org wrote&#8203;:
> > A Bug ID \(20001105\.014\) has been assigned and is shown in the subject of this email\.  
> 
> Unexpectedly\, perl\.org passed my message thru when it shouldn't have\. Please
> disregard/close/kill the previous bug report\, as it was obviously bogus\.

Closed 20001105\.014\.

**Comment by p5pRT** [comment]
### From @jhi

On Mon\, Nov 06\, 2000 at 04&#8203;:07&#8203;:46PM \+0100\, H\.Merijn Brand wrote&#8203;:
> On Mon\, 6 Nov 2000 00&#8203;:11&#8203;:14 \+0100\, Marc Lehmann \<pcg@&#8203;goof\.com> wrote&#8203;:
> > On Sun\, Nov 05\, 2000 at 11&#8203;:09&#8203;:19PM \-0000\, perlbugtron@&#8203;bugs\.perl\.org wrote&#8203;:
> > > A Bug ID \(20001105\.014\) has been assigned and is shown in the subject of this email\.  
> > 
> > Unexpectedly\, perl\.org passed my message thru when it shouldn't have\. Please
> > disregard/close/kill the previous bug report\, as it was obviously bogus\.
> 
> Closed 20001105\.014\.

Uhh\, I prefer the other way round\.\.\. I reopened the 1105\.014 since
it's by Marc\, and closed the \.013 which was my forward of the same
as a duplicate\.
